src/CMakeLists.txt

```diff
@@ -512,40 +609,54 @@ if(CMAKE_CROSSCOMPILING)
       target_compile_options(${elf} PRIVATE --specs=rdimon.specs)
       target_link_libraries(${elf} PRIVATE --specs=rdimon.specs)
     endif()
+    string(FIND ${elf} bitboxbase BITBOXBASE_FOUND)
+    if(BITBOXBASE_FOUND EQUAL -1)
+      target_sources(${elf} PRIVATE ${BITBOX02-FIRMWARE-SOURCES})
+    else()
+      target_sources(${elf} PRIVATE ${BITBOXBASE-FIRMWARE-SOURCES})
+    endif()
```

Comment:
I don't think this pattern (find string in elf name) works well here. It works well for the "semhosting" string because it is binary. I think it would be better if you created two variables: "FIRMWARES-BITBOX02" and "FIRMWARE-BITBOXBASE" a bit like there is for the bootloaders. Then you could iterate only over the bb02 firmwares or the bbb firmwares and add sources.

Reply:
Fixed.